modules/apigee_edge_teams/src/TeamContextManagerInterface.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/**
+ * Copyright 2018 Google Inc.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * version 2 as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA 02110-1301, USA.
+ */
+
+namespace Drupal\apigee_edge_teams;
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Url;
+
+/**
+ * Base definition of the team context manager service.
+ */
+interface TeamContextManagerInterface {
```

Comment:
I was thinking this too since this implementation is pretty specific to providing for the team switcher but at some point I am hoping to get authoritative source for the question. "Is the route I'm on for developers or is it for teams". This can be determined by using `getCurrentContext` and checking the entity type. Maybe in the future we have `getCurrentContext` return `static::CONTEXT_TYPE_TEAM` or `CONTEXT_TYPE_DEVELOPER` so I think it makes sense to rename `getCurrentContext` to `getCurrentContextEntity` or something like that.